app/models/host/managed.rb

```diff
@@ -806,6 +815,33 @@ def setup_clone
     @old = super { |clone| clone.interfaces = self.interfaces.map {|i| setup_object_clone(i) } }
   end
 
+  def refresh_global_status
+    self.global_status = build_global_status.status
+  end
+
+  def refresh_statuses
+    HostStatus.status_registry.each do |status_class|
+      status = get_status(status_class.to_s)
+      status.refresh! if status.relevant?
+    end
+    host_statuses.reload
+    refresh_global_status
+  end
+
+  def get_status(type)
+    status = host_statuses.find_by_type(type)
+    if status.nil?
+      status_class = type.constantize
```

Comment:
I find this string-based API slightly strange, would it be more natural to pass the class/constant itself?